### ISSUE TITLE
Backport #170 (IoC improvements) into existing release

### ISSUE THREAD

**Issue by mgoodfellow** [TurnerSoftware/CacheTower#174]
Hi,

I notice master has been stuck on this custom serialiser work for a while, and in addition I see you are now stuck with a protobuf change that doesn't seem to be getting much traction over there. As I believe(?) we have had a partial merge of some of the customer serialiser work, master is effectively blocked.

Would you be happy to do a 0.9.1 release, or a 0.10.0 release (I guess 0.10.0 is more relevant from a semver perspective) where we fork from pre-master (we can fork the 0.9 tag) and cherry pick my #170 PR into it? (and I guess anything else that might be wanted in a fix release)

I'm happy to prep it as a branch, but I don't want to mess up your future release plans?

Many thanks!

Mike

**Comment by mgoodfellow** [comment]
I've opened the PR to revert the changes to master - it wasn't as messy as I thought it would be - which leads me to wonder if it would be better to remove the code from master, then open a new PR to add it back in when it has been unblocked, rather than maintaining a hotfix branch against a hotfix release?

**Comment by Turnerj** [comment]
Yeah, you're right. I really hoped to get the serialization stuff done but with the roadblock due to Protobuf (and no response about how, if I raised a PR, I should build it), what you've suggested seems the best way forward.

I'll take a careful look at your PR and work out how to progress getting back to the pre-serialization merge state.

**Comment by Turnerj** [comment]
New release of Cache Tower is out (v0.10.0) and has the IoC improvements and a few other minor things. Let me know how it goes and if you hit any issues with it 🙂

**Comment by mgoodfellow** [comment]
Awesome thanks @Turnerj! I will give this a spin this coming week :) (background refresh here I come!)

**Comment by mgoodfellow** [comment]
Btw I chucked together a nuget for SimpleInjector - I think I'm regretting the name now, probably should have been SimpleInjectorActivator? 👀 

https://github.com/mgoodfellow/CacheTower.SimpleInjectorContext

I'm consuming it at the moment, but obviously pending the dep fixes before I issue as stable to finish testing

**Comment by Turnerj** [comment]
That's awesome!

I guess while you are the only consumer, you might still be able to change the name without much fuss - then just unlist the existing package on NuGet. I changed the name of a package of mine because the company it related to changed their name - that was a little messy.

**Comment by mgoodfellow** [comment]
Well, I think I've flipped back the other way, and now think the name fits.

All that being said - what do you think? Ideally a pattern should be kept for IoC container / package name for extension.

Happy to take any feedback and change name accordingly.

Another thought to keep existing patterns would be something like:

CacheTower.Contexts.SimpleInjectorActivator

?

**Comment by Turnerj** [comment]
I was thinking about that and yeah, might be a good idea to establish a pattern. Because it is a "Cache Context Activator", maybe: `CacheTower.ContextActivators.SimpleInjector`

That follows the rough pattern I have for extensions and providers too. Thoughts?

**Comment by mgoodfellow** [comment]
I like it! I will update the package shortly and make the change :)

Thanks!

**Comment by mgoodfellow** [comment]
All done!

https://github.com/mgoodfellow/CacheTower.ContextActivators.SimpleInjector
https://www.nuget.org/packages/CacheTower.ContextActivators.SimpleInjector/
